### PR TITLE
feat(reviewer): strengthen code-review skill for intent drift, regression risk, and side effects

### PR DESCRIPTION
**Pull request by devxoul**
## Background

The `code-review` skill previously had three weak spots that let a class of real review failures through undetected:

1. **Architecture fit checked only local structure.** The "Architecture fit" item caught layering and duplication but didn't name *intent drift* — a diff that compiles, passes its own tests, and still pulls the codebase away from the design intent the surrounding code was built on. A "purely a refactor" that quietly changes observable behavior, an "internal only" change that shifts an exported contract, a temporary branch that bypasses an established abstraction: these all look locally correct and slip through a checklist that only asks about layering.

2. **No regression concept beyond deleted tests.** There was no first-class treatment of blast radius — the existing consumers that silently break because a contract the diff changes was one they depended on. The regression never appears in the diff or this PR's test count; it lives in a caller the reviewer has to go find.

3. **Side effects were uncovered.** Shared/global state mutation, un-invalidated caches, downstream consumers of a changed event/log/metric shape, migrations that must run in lockstep, resources leaked on a new error path: these have no line in the diff to anchor to and only emerge from interaction with the surrounding code.

## Summary

**`src/bundled-plugins/reviewer/skills/code-review.ts`**

- Renamed item 3 to **"Architecture fit and intent drift"** and extended it to describe the drift pattern: a change that is locally valid but diverges from the stated or implicit intent (e.g., a description that claims "no behavior change" when the diff does change behavior). The item now instructs the reviewer to verify descriptions hold their stated line.
- Added item 4 **"Regression risk and blast radius"**: grep callers of changed exports, trace who depended on the old behavior, state the blast radius explicitly. Counters the false clearance of "looks fine in the diff."
- Added item 5 **"Side effects and ripple"**: shared state, cache invalidation, event/metric shape, lockstep migrations, cleanup-path coverage on new error branches. Instructs the reviewer to trace the interaction, not just the changed lines.
- Updated the severity-hint section so regressions you can name and side effects (leaked resource, un-invalidated cache, mutated shared state) map to **blocker**, while suspected-but-unconfirmed blast radius maps to **concern** with an explicit "say what you'd check" nudge.

**`src/bundled-plugins/reviewer/reviewer.test.ts`**

Three drift-guard tests matching the existing convention: each asserts that load-bearing phrases survive future edits to the skill content.

## What this does NOT do

- Does not change the reviewer plugin's runtime behavior, tool surface, or any other skill.
- Does not alter the priority ordering — the three new items slot in at 4 and 5, pushing the original items 4–9 to 6–11.

## Review notes

The only load-bearing change is the string content of `code-review.ts`. The three new tests are the canonical regression guard for these additions — if the phrases are removed or renamed, CI fails.

## Tests

```
bun run typecheck     # clean
bun run lint          # 0 errors (66 pre-existing warnings in unrelated files)
bun run format:check  # clean
bun run test          # 7875 pass, 0 fail (3 new tests included)
```

### CODE REVIEW

File: src/bundled-plugins/reviewer/skills/code-review.ts [hunk 1]
```diff
@@ -61,13 +61,15 @@ Prioritize in this order:
...
+6. **Test coverage.** New behavior should have new tests. Edge cases the description names should be tested. If existing tests were deleted or skipped, that is a blocker absent a stated reason. Look past the raw test count, but only flag a redundant case when you can show the *inputs themselves* reach the same path — same branch, same validation rule, same boundary — not merely that the assertion shape is identical. Table-driven and parametrized tests legitimately share one assertion across many inputs while each input exercises a distinct branch, parser, or edge case; that is coverage, not duplication. The finding is "these inputs are indistinguishable to the code under test," and you must name the path they collapse onto — never "the assertions look the same."
```
[the remark this answers is not in the document]
Comment (devxoul) [reply src/bundled-plugins/reviewer/skills/code-review.ts:67]: Good catch — fixed in 92acd6a7. Reworded so the side-effect insight no longer contradicts the anchor rule: the *consequence* isn't obvious from the changed line, but there's still a line that introduces it, so the finding now anchors to the mutation/lifecycle/config line where the ripple enters and names the downstream consumer that breaks.
